src/main/java/org/entando/entando/web/filebrowser/FileBrowserController.java

```diff
@@ -99,12 +99,24 @@ public ResponseEntity<RestResponse<Map<String, Object>, Map<String, Object>>> ge
             @RequestParam(value = CURRENT_PATH, required = false, defaultValue = "") String currentPath,
             @RequestParam(value = PROTECTED_FOLDER, required = false, defaultValue = "false") Boolean protectedFolder) {
         logger.debug("required file {} - protected {}", currentPath, protectedFolder);
-        byte[] base64 = this.getFileBrowserService().getFileStream(currentPath, protectedFolder);
+
+        boolean isDirectory = this.getFileBrowserService().isDirectory(currentPath, protectedFolder);
+
         Map<String, Object> result = new HashMap<>();
         result.put(PROTECTED_FOLDER, protectedFolder);
+
+        result.put("isDirectory", isDirectory);
         result.put("path", currentPath);
-        result.put("filename", this.getFilename(currentPath));
-        result.put("base64", base64);
+
+        if (!isDirectory){
+            byte[] base64 = this.getFileBrowserService().getFileStream(currentPath, protectedFolder);
+            result.put("filename", this.getFilename(currentPath));
+            result.put("base64", base64);
+        } else {
+            result.put("filename", "");
```

Comment:
Maybe we could populate this for the directory too, what do you think?

Reply:
Hi @zonia3000 thanks for the review. 
I'm not sure about this change because this is the directory case so we don't have a file name. 
@cecchisandrone WDYT?

Reply:
Empty value could be worse. Probably better having the directory name here, even if probably not used by the client

Reply:
done